node-notes-cli/app.js

```diff
@@ -0,0 +1,43 @@
+const fs = require('fs');
+fs.readFile('data.json', 'utf8', (err, data) => {
+  if (err) throw err;
+
+  var obj = JSON.parse(data);
+
+  if (process.argv[2] === 'read') {
+
+    for (const prop in obj.notes) {
+      console.log(`${prop}: ${obj.notes[prop]}`);
+    }
+
+  }
+
+  if (process.argv[2] === 'create') {
+
+    var id = obj.nextId;
+    obj.notes[id] = process.argv[3]
+    obj.nextId++;
+
+  }
+
+  if (process.argv[2] === 'update') {
+
+    var num = process.argv[3];
+    obj.notes[num] = process.argv[4];
+
+  }
```

Comment:
"User can update a note" shouldn't be able to create a new note by accident, only modify existing ones. How would you prevent accidental note creation?

Reply:
Updated. Add one more condition.